CMakeLists.txt

```diff
@@ -1,4 +1,9 @@
 cmake_minimum_required(VERSION 3.16)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON) # does not produce the json file
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE INTERNAL "") # works
+
+set(CMAKE_OSX_ARCHITECTURES "x86_64")
```

Comment:
This does not belong into the CMakeLists.txt you may create a toolchain file with this info or pass it on the command line.

Reply:
Can you explain how to create this toolchain file? I'm very new to CMake

Reply:
Can you explain how to create this toolchain file? I'm very new to CMake